VSRAD.DebugServer/Handlers/PutFileHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using VSRAD.DebugServer.IPC.Commands;
+using VSRAD.DebugServer.IPC.Responses;
+
+namespace VSRAD.DebugServer.Handlers
+{
+    public sealed class PutFileHandler : IHandler
+    {
+        private readonly PutFileCommand _command;
+
+        public PutFileHandler(PutFileCommand command)
+        {
+            _command = command;
+        }
+
+        public Task<IResponse> RunAsync()
+        {
+            var fullPath = Path.Combine(_command.WorkDir, _command.Path);
+            var status = DoPutFile(fullPath, _command.Data);
+            return Task.FromResult<IResponse>(new PutFileResponse { Status = status });
```

Comment:
This method is not really async. In fact, it just creates a completes task with a result. You may want to use something like this: [TaskFactory.StartNew](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskfactory.startnew?view=netcore-3.1) If you want to start some actual work in a new task.

Reply:
Fixed it, thanks!